django_s3_storage/storage.py

```diff
@@ -92,34 +123,81 @@ def open(self, mode="rb"):
         return super().open(mode)
 
 
-class _Local(local):
+@dataclass
+class Settings:
+    # default_auth_settings
+    AWS_REGION: str = "us-east-1"
+    AWS_ACCESS_KEY_ID: str = ""
+    AWS_SECRET_ACCESS_KEY: str = ""
+    AWS_SESSION_TOKEN: str = ""
+    # default_s3_settings
+    AWS_S3_ADDRESSING_STYLE: str = "auto"
+    AWS_S3_ENDPOINTS: dict = field(
+        default_factory=lambda: {
+            's3': Endpoints(),
+        }
+    )
+    AWS_S3_KEY_PREFIX: str = ""
+    AWS_S3_MAX_AGE_SECONDS: int = 60 * 60  # 1 hours.
+    AWS_S3_REDUCED_REDUNDANCY: bool = False
+    AWS_S3_CONTENT_DISPOSITION: str = ""
+    AWS_S3_CONTENT_LANGUAGE: str = ""
+    AWS_S3_METADATA: dict = field(default_factory=dict)
+    AWS_S3_ENCRYPT_KEY: bool = False
+    AWS_S3_KMS_ENCRYPTION_KEY_ID: str = ""
+    AWS_S3_GZIP: bool = True
+    AWS_S3_SIGNATURE_VERSION: str = "s3v4"
+    AWS_S3_FILE_OVERWRITE: bool = False
+    AWS_S3_USE_THREADS: bool = True
+    AWS_S3_MAX_POOL_CONNECTIONS: int = 10
+    AWS_S3_CONNECT_TIMEOUT: int = 60  # 60 seconds
+    AWS_S3_BUCKET_NAME: str = 'Deprecated'
+    AWS_S3_READ_ONLY: bool = False
```

Comment:
Keep order from original settings, i.e. move `AWS_S3_BUCKET_NAME` back up

```suggestion
    # default_s3_settings
    AWS_S3_BUCKET_NAME: str = 'Deprecated'
    AWS_S3_ADDRESSING_STYLE: str = "auto"
    AWS_S3_ENDPOINTS: dict = field(
        default_factory=lambda: {
            's3': Endpoints(),
        },
    )
    AWS_S3_KEY_PREFIX: str = ""
    AWS_S3_MAX_AGE_SECONDS: int = 60 * 60  # 1 hours.
    AWS_S3_REDUCED_REDUNDANCY: bool = False
    AWS_S3_CONTENT_DISPOSITION: str = ""
    AWS_S3_CONTENT_LANGUAGE: str = ""
    AWS_S3_METADATA: dict = field(default_factory=dict)
    AWS_S3_ENCRYPT_KEY: bool = False
    AWS_S3_KMS_ENCRYPTION_KEY_ID: str = ""
    AWS_S3_GZIP: bool = True
    AWS_S3_SIGNATURE_VERSION: str = "s3v4"
    AWS_S3_FILE_OVERWRITE: bool = False
    AWS_S3_USE_THREADS: bool = True
    AWS_S3_MAX_POOL_CONNECTIONS: int = 10
    AWS_S3_CONNECT_TIMEOUT: int = 60  # 60 seconds
    AWS_S3_READ_ONLY: bool = False
```